pkg/controller/infrastructure/infraflow/ensure.go

```diff
@@ -297,11 +415,33 @@ func (fctx *FlowContext) ensureFirewallRules(ctx context.Context) error {
 	}
 	vpc := GetObject[*compute.Network](fctx.whiteboard, ObjectKeyVPC)
 
-	cidrs := []*string{fctx.podCIDR, fctx.config.Networks.Internal, ptr.To(fctx.config.Networks.Workers), ptr.To(fctx.config.Networks.Worker)}
+	cidrs := []*string{
+		fctx.podCIDR,
+		fctx.config.Networks.Internal,
+		ptr.To(fctx.config.Networks.Workers),
+		ptr.To(fctx.config.Networks.Worker),
+	}
+
 	rules := []*compute.Firewall{
-		firewallRuleAllowInternal(firewallRuleAllowInternalName(fctx.clusterName), vpc.SelfLink, cidrs),
-		firewallRuleAllowHealthChecks(firewallRuleAllowHealthChecksName(fctx.clusterName), vpc.SelfLink),
+		firewallRuleAllowInternal(shared.FirewallRuleAllowInternalName(fctx.clusterName), vpc.SelfLink, cidrs),
+		firewallRuleAllowHealthChecks(shared.FirewallRuleAllowHealthChecksName(fctx.clusterName), vpc.SelfLink, healthCheckSourceRangesIPv4),
+	}
+
+	cidrsipv6 := []*string{}
+	if nodesipv6 := fctx.whiteboard.Get(NodesSubnetIPv6CIDR); ptr.Deref(nodesipv6, "") != "" {
+		cidrsipv6 = append(cidrsipv6, nodesipv6)
+	}
+	if servicesipv6 := fctx.whiteboard.Get(ServicesSubnetIPv6CIDR); ptr.Deref(servicesipv6, "") != "" {
+		cidrsipv6 = append(cidrsipv6, servicesipv6)
+	}
```

Comment:
To stays consistent we should use camel case variable naming in `cidrsipv6` = `cidrsIPv6`, `nodesipv6` = `nodesIPv6` and `servicesipv6` = `servicesIPv6`